### DISCUSSION TITLE
Lack of control with PushSubscribeAsync

### DISCUSSION THREAD

**Discussion by SkinnySackboy** [nats-io/nats.net.v1#722] [Q&A]
Hi,

We have got the following code, which (minus the last line which appears not to do anything) works.

Is there any way for me to call PushSubscribeAsync to create the subscription, but to maintain control over when I can call start, rather than having the subscription start receiving messages immediately?

```csharp
var pushSubscribeOptions = new PushSubscribeOptionsBuilder()
                                .WithDurable(Guid.NewGuid().ToString())
                                .Build();

var asyncSubscription = m_jetStreamContext.PushSubscribeAsync(subject, NATSSubscriptionHandler, true, pushSubscribeOptions);
asyncSubscription.Start(); // This line does not nothing, as the preceding line starts immediately
```

 I tried checking if there is some manual mode I can opt in to but I cannot see anything obvious that I'm missing.

Thanks in advance.

**Comment by scottf** [comment]
Start() is called for you. I think you maybe used to recent .net async versus what this client actually does. But for now, once the consumer and subscription exist, the server starts pushing data.
You could create the consumer ahead of time with the JetstreamManagement (or the CLI) and simply subscribe when you are ready using the bind functionality.

**Comment by SkinnySackboy** [comment]
Thanks. Do you think it could make sense in the future to support something like this?
```csharp
var pushSubscribeOptions = new PushSubscribeOptions.PushSubscribeOptionsBuilder()
                                    .WithManualStart(true) // This would allow me to call Start when I decide. Default is false for backward compatibility
                                    .Build();

var asyncSubscription = m_jetStreamContext.PushSubscribeAsync(subject, NATSSubscriptionHandler, true, pushSubscribeOptions);

// Now I can call this 
asyncSubscription.Start();
```
